localstack/utils/bootstrap.py

```diff
@@ -41,16 +41,43 @@
 
 LOG = logging.getLogger(__name__)
 
-# maps from API names to list of other API names that they depend on
+# Mandatory dependencies of services on other services
+# - maps from API names to list of other API names that they _explicitly_ depend on: <service>:<dependent-services>
+# - an explicit service dependency is a service without which another service's basic functionality breaks
+# - this mapping is used when enabling strict service loading (use SERVICES env var to allow-list services)
+# - do not add "optional" dependencies of services here, use API_DEPENDENCIES_OPTIONAL instead
 API_DEPENDENCIES = {
     "dynamodb": ["dynamodbstreams"],
+    # dynamodbsteams uses kinesis under the hood
     "dynamodbstreams": ["kinesis"],
+    # es forwards all requests to opensearch (basically an API deprecation path in AWS)
     "es": ["opensearch"],
     "cloudformation": ["s3", "sts"],
-    "lambda": ["s3", "sqs", "sts"],
+    "lambda": ["s3", "sts"],
```

Comment:
I moved the "sqs" dependency to optional here with 7f7f143832afed217261a57d86f5ce72aa231c34, since #10062 it is not required anymore for the internal behavior of lambda per default.

However due to event source mapping / destination / dead letter queues, it is still an optional dependency.

Reply:
No worries, thanks for changing the deps! 🚀